### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by vapelavsky**
Buttons with a little bit of magic
Admin can describe why he declined bug
Messages which set on UI/UX standards

### CODE REVIEW

File: bot/chains/register_bug/handlers.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 
 @dp.callback_query_handler(lambda x: x.data == 'bug_register_cancel', state='*')
 async def cancel(c: types.CallbackQuery, state: FSMContext):
+    await bot.edit_message_reply_markup(c.from_user.id, c.message.message_id, reply_markup='')
```
Comment (Mo1far) [inline bot/chains/register_bug/handlers.py]: replace to cq.message.delete_reply_markup()

File: bot/chains/register_bug/handlers.py [hunk 2]
```diff
@@ -28,6 +29,7 @@ async def add_bug_start(msg: types.Message):
 
 @dp.message_handler(state=RegisterBug.wait_photo, content_types=['photo', 'video', 'document'])
 async def add_bug_photo(msg: types.Message, state: FSMContext):
+    await bot.edit_message_reply_markup(msg.chat.id, message_id=msg.message_id - 1, reply_markup='')
```
Comment (Mo1far) [inline bot/chains/register_bug/handlers.py]: replace to cq.message.delete_reply_markup()

File: bot/chains/register_bug/handlers.py [hunk 3]
```diff
@@ -39,13 +41,15 @@ async def add_bug_photo(msg: types.Message, state: FSMContext):
 
 @dp.message_handler(state=RegisterBug.wait_description)
 async def add_bug_description(msg: types.Message, state: FSMContext):
+    await bot.edit_message_reply_markup(msg.chat.id, message_id=msg.message_id - 1, reply_markup='')
```
Comment (Mo1far) [inline bot/chains/register_bug/handlers.py]: replace to cq.message.delete_reply_markup()

File: bot/chains/register_bug/handlers.py [hunk 4]
```diff
@@ -39,13 +41,15 @@ async def add_bug_photo(msg: types.Message, state: FSMContext):
...
     await state.update_data({'description': msg.text})
     await RegisterBug.wait_location.set()
     await msg.answer('Надішліть місцезнаходження (аудиторію, корпус) якомога конкретніше 🏢', reply_markup=cancel_kb)
 
 
 @dp.message_handler(state=RegisterBug.wait_location)
 async def add_bug_location(msg: types.Message, state: FSMContext):
+    await bot.edit_message_reply_markup(msg.chat.id, message_id=msg.message_id - 1, reply_markup='')
```
Comment (Mo1far) [inline bot/chains/register_bug/handlers.py]: replace to cq.message.delete_reply_markup()

File: bot/chains/register_bug/handlers.py [hunk 5]
```diff
@@ -70,22 +74,30 @@ async def add_bug_location(msg: types.Message, state: FSMContext):
...
-        await bot.send_message(bug.user, f'Мабуть ти помилився при описі багу № {bug.id} 😔\n\n'
-                                         'Наразі відхиляємо твоє повідомлення, '
-                                         'але ти можеш додати трішки кращий опис та спробувати ще раз 😉')
-
+        await bot.edit_message_caption(ADMIN_CHAT_ID, message_id=cq.message.message_id, caption=cq.message.caption +
+                                                                                                '\n\nБаг відхилено ❌', )
+        await bot.send_message(ADMIN_CHAT_ID, 'Опишіть, чому цей баг відхилено 🤔')
+        await RegisterBug.wait_cause.set()
+
+        @dp.message_handler(state=RegisterBug.wait_cause)
```
Comment (Mo1far) [inline bot/chains/register_bug/handlers.py]: What? handler in another handler? How you can explain this?

File: bot/chains/register_bug/handlers.py [hunk 6]
```diff
@@ -70,22 +74,30 @@ async def add_bug_location(msg: types.Message, state: FSMContext):
...
+        async def cause_text(msg: types.Message, state: FSMContext):
+            await state.set_data({'cause': msg.text})
```
Comment (Mo1far) [inline bot/chains/register_bug/handlers.py]: It's useless, no?

File: bot/chains/register_bug/handlers.py [hunk 7]
```diff
@@ -70,22 +74,30 @@ async def add_bug_location(msg: types.Message, state: FSMContext):
...
+            data = await state.get_data()
+            await bot.send_message(ADMIN_CHAT_ID, f"Дякуємо! Причина буде передана відправнику!")
+            await bot.send_message(bug.user, f"Баг № {bug.id} відхилено 😔\n\nПричина: {data.get('cause')}")
+            await bug.update(cause=data.get('cause')).apply()
```
Comment (Mo1far) [inline bot/chains/register_bug/handlers.py]: useless ?

File: bot/chains/register_bug/state.py [hunk 8]
```diff
@@ -5,3 +5,4 @@ class RegisterBug(StatesGroup):
     wait_photo = State()
     wait_location = State()
     wait_description = State()
+    wait_cause = State()
```
Comment (Mo1far) [inline bot/chains/register_bug/state.py]: maybe rename to the more describing variable?

File: db/models/bug.py [hunk 9]
```diff
@@ -24,3 +24,9 @@ class Bug(TimedBaseModel):
     location = db.Column(db.String(50))
     status = db.Column(db.Integer, db.ForeignKey('bug_status.id'), nullable=False)
     user = db.Column(db.Integer, db.ForeignKey('users.id'))
+    cause = db.Column(db.Text)
+
+    @classmethod
```
Comment (Mo1far) [inline db/models/bug.py]: overkill method
we can simply use User.get(id)

**Review by Mo1far — CHANGES_REQUESTED** [review]

File: bot/chains/register_bug/state.py [hunk 10]
```diff
@@ -5,3 +5,6 @@ class RegisterBug(StatesGroup):
     wait_photo = State()
     wait_location = State()
     wait_description = State()
+    wait_admin_description = State()
+    message = State()
+    bug = State()
```
Comment (Mo1far) [inline bot/chains/register_bug/state.py]: useless ?

File: bot/chains/register_bug/handlers.py [hunk 11]
```diff
@@ -15,38 +15,48 @@
...
+async def add_bug_start(msg: types.Message, state: FSMContext):
     await RegisterBug.wait_photo.set()
-    await msg.answer('Надішліть фото багу 📸', reply_markup=cancel_kb)
+    answer = await msg.answer('Надішліть фото багу 📸', reply_markup=cancel_kb)
+    await state.update_data({'message': answer})
 
 
 @dp.message_handler(state=RegisterBug.wait_photo, content_types=['photo', 'video', 'document'])
 async def add_bug_photo(msg: types.Message, state: FSMContext):
+    message = await state.get_data()
```
Comment (Mo1far) [inline bot/chains/register_bug/handlers.py]: state_data in var message right? maybe rename to state_data, etc ?

File: bot/chains/register_bug/handlers.py [hunk 12]
```diff
@@ -69,23 +79,33 @@ async def add_bug_location(msg: types.Message, state: FSMContext):
...
+                                       message_id=cq.message.message_id,
+                                       caption=cq.message.caption + '\n\nБаг відхилено ❌')
+
+        await bot.send_message(ADMIN_CHAT_ID, 'Опишіть, чому цей баг відхилено 🤔')
+        await RegisterBug.wait_admin_description.set()
+        await state.set_data({'bug': bug})
 
     await bug.update(status=status).apply()
 
-    await cq.answer('Готово')
```
Comment (Mo1far) [inline bot/chains/register_bug/handlers.py:91]: maybe we can send cq answer from the different decision?

**Review by Mo1far — CHANGES_REQUESTED** [review]

File: bot/chains/register_bug/handlers.py (hunk 12 above)
Comment (vapelavsky) [reply bot/chains/register_bug/handlers.py:91]: Ты плюс с минусом перепутал, онгличанин

**Review by Mo1far — CHANGES_REQUESTED** [review]
rename variables and check state usage

**Review by Mo1far — APPROVED** [review]

**Pull request merged**